rfcs/008-public-api.md

```diff
@@ -0,0 +1,90 @@
+---
+feature: stable public API
+start-date: 2025-09-05
+author: Joe Neeman
+---
+
+# A stable API for Nickel
+
+Currently, the only stable interface to Nickel is the CLI tool; anyone
+wanting to embed the Nickel interpreter needs to deal with the
+`nickel_lang_core` crate, which changes frequently and isn't
+particularly user-friendly.
+
+This RFC proposes a simple Nickel API. Its main goals are
+
+- Simplicity: it should be easy to do basic things, like evaluating
+  some Nickel code into a structured representation.
+- Stability: we aim not to break public APIs.
```

Comment:
I wonder if we shouldn't tame this down a bit. Do we really want to promise from now that the API will stay stable forever? It does sound a bit strong to me. However, we could say that the goal is to make breaking changes as rare as possible?